File: 69 Sqrt(x)/solution.cpp
class Solution {
public:
    int mySqrt(int x) {
        if (x==0)
            return 0;
        long helper = 0,result = 0;
        while ((result+1)*(result+1)<=x){
            helper = 1;
            while ((result+helper)*(result+helper)<=x){
                helper = helper*2;
            }
            result+=helper/2;
        }
        return result;
    }
};

File: 69 Sqrt(x)/solution.py
class Solution:
    # @param {integer} x
    # @return {integer}
    def mySqrt(self, x):
        if x==0:
            return 0
        tmp,helper = 1,0
        result,getin = 0,0
        while x>=(result+tmp)*(result+tmp):
            helper = tmp
            while x>=(result+helper)*(result+helper):
                getin = helper
                helper*=2
            result+=getin
        return result